mapping_parenting_tech/analysis/prototyping/play_store/get_play_store_apps.py

```diff
@@ -7,141 +7,670 @@
+        for app_id in csv_reader:
+            app_id_list.extend(app_id)
+
+    app_category = app_id_list.pop(0)
+    return (app_category, app_id_list)
 
-logging.info(
-    f"There are {review_counter} reviews available for {len(output_ids['Parenting apps'])} apps. {len(remove_apps)} apps were removed"
-)
-logging.info(
-    f"App id {most_reviews['id']} had the most reviews with {most_reviews['count']}"
-)
 
 # %% [markdown]
-# Save app details
+# ## Retrieving app details
+# These functions get details for a given list of apps, specified by their ids (as retrieved above) and save and load those details.
+#
+# `get_playstore_app_details` takes a list of app ids and returns the details for those apps as a list of `dict` objects. The schema for each `dict` can be seen on [https://pypi.org/project/google-play-scraper/](https://pypi.org/project/google-play-scraper/)
+#
+# `save_app_details` saves the list of `dicts` retrieved by `get_playstore_app_details`
+#
+# `load_app_details` loads the app details saved by `save_app_details`
+
+# %%
+def get_playstore_app_details(app_id_list: list):
+    """
+    Uses `google-play-scraper` (https://pypi.org/project/google-play-scraper/) to retrieve details about apps given in
+    `app_id_list` and returns dict of app details
+
+    Args:
+        app_id_list: list - a list of app ids from which details will be retrieved
+
+    Returns:
+        List of dict objects with app details. Each dict includes:
+        - title
+        - appId
+        - description
+        - summary
+        for specifics of all data returned, see https://pypi.org/project/google-play-scraper/
+
+    """
+
+    all_app_details = list()
+    remove_apps = list()
+
+    for app_id in tqdm(
+        app_id_list,
+        desc="Retrieving app details",
+    ):
+        try:
+            app_details = app(app_id, country="gb")
+            all_app_details.append({app_id: app_details})
+
+        except Exception as e:  # needs modifying to capture specific errors
+            logging.warning(f"Error on app id {app_id}: {e} {repr(e)}")
+            remove_apps.append(app_id)
+
+    for app_id in remove_apps:
+        all_app_details.pop(app_id, None)
+
+    return all_app_details
+
+
+# %%
+def save_app_details(
+    details_dict: dict, filename: str, folder: str = APP_IDS_PATH
+) -> bool:
+    """
+    Saves JSON representation of app details. Will append to file <filename> if it already exists.
+    Returns True if exits successfully
+
+    Args:
+        details_dict: dict - a dictionary of app details
+        filename: str - name of the file to save the details to
+        folder: str, default is `APP_IDS_PATH` - name of the folder where `filename` is to be saved
+
+    Returns:
+        True if successful
+
+    """
+
+    output_target = folder / filename
+    with open(output_target, "at") as output_file:  # append in text mode
+        json.dump(details_dict, output_file, indent=2, default=str)
+
```

Comment:
Same consideration as above:
```suggestion

    logging.info(f"App details saved in {output_target}")
```